### PR TITLE
Fixed #34547 -- Deprecated DatabaseOperations.field_cast_sql().

### PR DESCRIPTION
**Pull request by smithdc1**
[Ticket #34547](https://code.djangoproject.com/ticket/34547)

### CODE REVIEW

**Comment by felixxm** [comment]
@smithdc1 Do you have time to keep working on this?

File: django/db/models/lookups.py [hunk 1]
```diff
@@ -217,8 +217,6 @@ class BuiltinLookup(Lookup):
     def process_lhs(self, compiler, connection, lhs=None):
         lhs_sql, params = super().process_lhs(compiler, connection, lhs)
         field_internal_type = self.lhs.output_field.get_internal_type()
-        db_type = self.lhs.output_field.db_type(connection=connection)
-        lhs_sql = connection.ops.field_cast_sql(db_type, field_internal_type) % lhs_sql
```
Comment (felixxm) [inline django/db/models/lookups.py:221]: We should still call `field_cast_sql()` when it's implemented in the 3rd-party database backends and add a deprecation warning here, e.g.
```python
# RemovedInDjango60Warning.
if connection.ops.field_cast_sql is not BaseDatabaseOperations.field_cast_sql:
    warnings.warn(...)
    db_type = self.lhs.output_field.db_type(connection=connection)
    lhs_sql = connection.ops.field_cast_sql(db_type, field_internal_type) % lhs_sql
```

Comment (felixxm) [reply django/db/models/lookups.py:221]: @smithdc1 I can add an extra warning if you're busy :point_up:

Comment (smithdc1) [reply django/db/models/lookups.py:221]: Yes please. 🎁

**Comment by felixxm** [comment]
@smithdc1 Thanks :+1: I pushed final edits.

**Pull request merged**